File: sample/spring-boot-jaxrs/src/main/java/com/benjaminsproule/sample/springboot/mvc/SampleResource.java
package com.benjaminsproule.sample.springboot.mvc;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.stereotype.Controller;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;

@Controller
@Path("/sample")
@Produces("application/json")
@Api(value = "/sample", description = "Sample REST for Integration Testing")
public class SampleResource {
    @GET
    @ApiOperation(value = "Return hello message", response = String.class)
    public String home() {
        return "{\"Hello\": \"World!\"}";
    }
}

File: sample/scala-spring-boot-jaxrs/src/main/scala/com/benjaminsproule/sample/scala/SampleResource.scala
package com.benjaminsproule.sample.scala

import io.swagger.annotations.{Api, ApiOperation}
import javax.ws.rs.{GET, Path, Produces}
import org.springframework.stereotype.Controller

@Controller
@Path("/sample")
@Produces(Array("application/json"))
@Api(value = "/sample", description = "Sample REST for Integration Testing")
class SampleResource {
    @GET
    @ApiOperation(value = "Return hello message", response = classOf[String])
    def home: String = {
        "{\"Hello\": \"World!\"}"
    }
}
